File: daily/11. 盛最多水的容器.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left=0,right=height.size()-1;
        int max=0;
        while(left<right){
            if((right-left)*min(height[left],height[right])>max)
                max=(right-left)*min(height[left],height[right]);
            else
                height[left]<height[right]?left++:right--;	
            
        }
        return max;
    }
};

File: daily/双指针/双指针medium/11. 盛最多水的容器.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var left=0,right=height.length-1,max=0;
    while(left<right)
    {
        if((right-left)*Math.min(height[left],height[right])>max)
			max=(right-left)*Math.min(height[left],height[right]);
		else
			height[left]<height[right]?left++:right--;	
    }
    return max;
};